src/libraries/System.Net.Primitives/src/System/Net/Cookie.cs

```diff
@@ -837,7 +837,7 @@ internal void ToString(StringBuilder sb)
             {
                 result += SeparatorLiteral + CookieFields.VersionAttributeName + EqualsLiteral + m_version.ToString(NumberFormatInfo.InvariantInfo);
             }
-            return result == EqualsLiteral ? null : result;
+            return result == EqualsLiteral.ToString() ? null : result;
```

Comment:
We should not be calling ToString here.  It's making things worse, not better :)

Reply:
Should the analyzer check to see if the value is passed to any other methods lacking a `char` overload? That seems potentially expensive for the analyzer to do--we'd likely want to limit the scan if we do.

Reply:
A fixer could introduce a new char const to be used, could just change the relevant call sites to use a literal char const, etc.

Reply:
The approach I was going for was to suggest a code-fix only when the const string being passed in is 1)a local variable or 2) a literal. For all the other cases, the analyzer still fires, but it's up to the dev to fix the code (which is what we're seeing in this snipped. EqualsLiteral is a class field, so there's no code-fix offered).

Reply:
Gotcha. Do you have any other suggestions for the analyzer, @pgovind, that could help other users avoid this trap of an accidental trade-off choice?

Reply:
I talked with @pgovind and we couldn't think of a succinct message that we could use on the analyzer to communicate to the user that they need to consider:

1. Converting a field or variable from `string` to `char` and then evaluating the trade-off between optimizations when the value is needed as the two different types
2. Converting a field or variable from to just be an inline literal
3. Introducing an additional field or variable so that you have one that is a `string` and one that is a `char`

Unless someone else has a suggestion for the analyzer's message that could help cover that, we'll leave it as "when applicable."

For this specific change, I recommended that we introduce a second `const` so that we have one that is a `string` and one that is a `char`, so that we avoid the introduction of inline literal strings or chars.